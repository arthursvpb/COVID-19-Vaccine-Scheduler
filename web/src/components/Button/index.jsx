/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Button, Typography } from '@material-ui/core';

import useStyles from './style';

export default function index({ children, ...otherProps }) {
  const classes = useStyles();

  return (
    <Button variant="outlined" className={classes.root} {...otherProps}>
      <Typography>{children}</Typography>
    </Button>
  );
}
