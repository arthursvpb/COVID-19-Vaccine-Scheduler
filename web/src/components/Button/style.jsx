import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    backgroundColor: '#d2f5e3',
    color: '#234435',
    '&:hover': {
      backgroundColor: '#b1dac5',
      color: '#234435',
    },
  },
});

export default useStyles;
