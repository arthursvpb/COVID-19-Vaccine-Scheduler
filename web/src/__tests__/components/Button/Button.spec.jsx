/* eslint-disable no-undef */
import React from 'react';

import { render } from '@testing-library/react';
import Button from '../../../components/Button';

test('renders correctly', () => {
  const { asFragment } = render(<Button />);

  expect(asFragment()).toMatchSnapshot();
});
