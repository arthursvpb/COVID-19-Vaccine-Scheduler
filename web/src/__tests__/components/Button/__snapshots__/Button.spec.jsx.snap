// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
